daemon/k8s/init.go

```diff
@@ -142,29 +141,3 @@ func WaitForNodeInformation(ctx context.Context, log logrus.FieldLogger, localNo
 	// want to specify them manually
 	return nil
 }
-
-// restoreRouterHostIPs restores (sets) the router IPs found from the
-// Kubernetes resource.
-//
-// Note that it does not validate the correctness of the IPs, as that is done
-// later in the daemon initialization when node.AutoComplete() is called.
-func restoreRouterHostIPs(n *nodeTypes.Node, log logrus.FieldLogger) {
-	if !option.Config.EnableHostIPRestore {
```

Comment:
The rationale of the commit which introduced the flag six years ago says it mainly exists so unit tests don't trip up if they are being run from a developer environment: https://github.com/giorio94/cilium/commit/a627e898067e65518a00306d795be34ce6272616

That was when most Cilium developers were running cilium-agent directly on the host, not within a container. I think that mode of development has fallen out of favor. Therefore, I'd say let's deprecate and remove it (assuming it passes CI).

Reply:
I double-checked the usages, and the relevant codepaths are already configuring a temporary state directory to prevent interference in case of leftover state. There would still be the possibility of issues in case a `cilium_host` interface exists locally, but my feeling is that there would be many more tests breaking if another Cilium instance was running on the same host in any case.

Given that, I've just adopted the easiest solution and dropped that config option altogether (we don't even need the deprecation process, as it was not configurable by users in any ways).

@gandro PTAL

Reply:
I didn't even realize the flag wasn't exposed via CLI flags. That makes the case for removal even easier.

Reply:
Yep, neither did I.